core/headerchain.go

```diff
@@ -458,6 +458,10 @@ func (hc *HeaderChain) loadLastState() error {
 			// properly and it doesn't crash the nodes
 			hc.currentHeader.Store(hc.genesisHeader)
 		}
+	} else {
+		// Recover the current header
+		log.Warn("Recovering Current Header")
+		hc.currentHeader.Store(hc.RecoverCurrentHeader())
```

Comment:
Do we also need to set the new head hash? i.e. so the next call to `ReadHeadBlockHash()` succeeds?